assets/composable/containerActions.ts

```diff
@@ -0,0 +1,47 @@
+type ContainerActions = "start" | "stop" | "restart";
+export const useContainerActions = () => {
+  const context = inject(containerContext);
+  if (!context) {
+    throw new Error("No container context provided");
+  }
+
+  const container = computed(() => context.container.value);
+  const { showToast } = useToast();
+
+  const actionStates = reactive({
+    stop: false,
+    restart: false,
+    start: false,
+  });
+
+  async function actionHandler(action: ContainerActions) {
+    const actionUrl = `/api/actions/${action}/${container.value.host}/${container.value.id}`;
+    const errors = {
+      404: "container not found",
+      500: "unable to complete action",
+      400: "invalid action",
+    } as Record<number, string>;
+    const defaultError = "something went wrong";
+
+    actionStates[`${action}`] = true;
+
+    try {
+      await fetch(withBase(actionUrl), { method: "POST" }).then((response) => {
```

Comment:
I think you missed my point earlier. Or maybe I am not getting it. But if you do `await` then you shouldn't need to do `.then(...)`.

Something like: 

```
const response = await fetch(...);
if(!response.ok){
...
}
```

Less code.

Reply:
Yup! I refactored:)